File: conformance/tests/inductive/inductive.runar.go
package contract

import "runar"

type Inductive struct {
	runar.InductiveSmartContract
	Count    runar.Int
	MaxCount runar.Int `runar:"readonly"`
}

func (c *Inductive) Increment(amount runar.Int) {
	c.Count = c.Count + amount
	runar.Assert(c.Count <= c.MaxCount)
}

func (c *Inductive) Reset() {
	c.Count = 0
}

File: conformance/tests/inductive/inductive.runar.rs
use runar::prelude::*;

#[runar::contract]
struct Inductive {
    count: Int,
    #[readonly]
    max_count: Int,
}

#[runar::methods(Inductive)]
impl Inductive {
    #[public]
    fn increment(&mut self, amount: Int) {
        self.count = self.count + amount;
        assert!(self.count <= self.max_count);
    }

    #[public]
    fn reset(&mut self) {
        self.count = 0;
    }
}
